File: VLC for iOSUITests/Helpers/XCUIElement+Helpers.m
/*****************************************************************************
 * XCUIElement+Helpers.m
 * VLC for iOS
 *****************************************************************************
 * Copyright (c) 2018 VideoLAN. All rights reserved.
 * $Id$
 *
 * Author: David Cordero <david # corderoramirez.com>
 *
 * Refer to the COPYING file of the official project for license.
 *****************************************************************************/

#import "XCUIElement+Helpers.h"

@implementation XCUIElement(Test)

- (void)clearAndEnterText:(NSString *)text {
    if (![[self value] isKindOfClass:[NSString class]]) {
        XCTFail("Tried to clear and enter text into a non string value");
        return;
    }

    [self tap];
    NSString *deleteString = @"";
    for (int i = 0; i < [(NSString *)[self value] length]; i++){
        deleteString = [deleteString stringByAppendingString:XCUIKeyboardKeyDelete];
    }

    [self typeText:deleteString];
    [self typeText:text];
}
@end

File: VLC-iOS-UITests/Helpers/XCUIElement+Helpers.swift
/*****************************************************************************
 * XCUIElement+Helpers.swift
 * VLC for iOSUITests
 *****************************************************************************
 * Copyright (c) 2018 VideoLAN. All rights reserved.
 * $Id$
 *
 * Authors: Mike JS. Choi <mkchoi212 # icloud.com>
 *
 * Refer to the COPYING file of the official project for license.
 *****************************************************************************/

import Foundation
import XCTest

extension XCUIElement {
    func clearAndEnter(text: String) {
        XCTContext.runActivity(named: "Enter \"\(text)\" into Textfield") { _ in
            guard let stringValue = self.value as? String else {
                XCTFail("Tried to clear and enter text into a non string value")
                return
            }
            
            tap()
            
            let deleteString = stringValue.map { _ in XCUIKeyboardKey.delete.rawValue }.joined(separator: "")
            typeText(deleteString)
            typeText(text)
        }
    }
}
